hmy/downloader/shortrange.go

```diff
@@ -51,21 +49,25 @@ func (d *Downloader) doShortRangeSync() (int, error) {
 	d.status.setTargetBN(expEndBN)
 	defer d.finishSyncing()
 
-	blocks, err := sh.getBlocksByHashes(hashChain, whitelist)
+	blocks, stids, err := sh.getBlocksByHashes(hashChain, whitelist)
 	if err != nil {
 		if !errors.Is(err, context.Canceled) {
 			sh.removeStreams(whitelist) // Remote nodes cannot provide blocks with target hashes
 		}
 		return 0, errors.Wrap(err, "getBlocksByHashes")
 	}
 
-	n, err := d.ih.verifyAndInsertBlocks(blocks)
+	n, err := verifyAndInsertBlocks(d.bc, blocks)
 	numBlocksInsertedShortRangeHistogramVec.With(d.promLabels()).Observe(float64(n))
 	if err != nil {
-		if !errors.As(err, &emptySigVerifyError) {
+		if sh.blameAllStreams(blocks, n, err) {
 			sh.removeStreams(whitelist) // Data provided by remote nodes is corrupted
+		} else {
+			// It is the last block gives a wrong commit sig. Blame the provider of the last block.
+			criminal := stids[len(stids)-1]
```

Comment:
criminal seems too strong , :-)

Reply:
Do you mean the variable name? Renamed to `stToBlame`. Thanks!